calico.yaml

```diff
@@ -0,0 +1,442 @@
+  target-architecture:
+    - x86_64
+    - aarch64
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - busybox
+      - ca-certificates-bundle
+      - build-base
+      - bash
+      - go
+      # for felix
+      - elfutils-dev
+      - zlib-dev
+      - iproute2
+      - zstd-dev
+      - llvm15
+      - clang-15
+      - libbpf
+      - linux-headers
+      # for BIRD
+      - flex
+      - bison
+      - autoconf
+      - automake
+      # TODO: Can remove this when cni-plugins are built from source
+      - curl
+
+# The calico build process is heavily tailored towards using Makefiles to
+# orchestrate docker commands. This makes for a less than ideal
+# experience replicating it in melange. To try to maintain some level of
+# sanity while packaging and updating calico, the approach here chooses to
+# use the upstream Makefile whenever possible, and only when necessary,
+# use the equivalent evaluated `docker run ...` commands within
+# independent pipeline blocks. Throughout this file you'll see the relevant
+# comments denoting the equivalent Makefile targets that each pipeline
+# block is trying to match, if applicable.
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/projectcalico/calico
+      tag: v${{package.version}}
+      expected-commit: 8b103f46fbdc989e59d81e08d215ab4a59fa6cec
+  - runs: |
+      # Remediate GHSA-hqxw-f8mx-cpmw
+      go get github.com/docker/distribution@v2.8.2-beta.1
+      go mod tidy
+  - working-directory: felix
+    pipeline:
+      # Equivalent to target: "build-bpf"
+      - runs: |
+          # TODO: this is a terrible hack to prevent the error:
+          #   /usr/include/gnu/stubs.h:7:11: fatal error: 'gnu/stubs-32.h' file not found
+          # I have no idea why it's wanting stubs-32.h and not correctly recognizing stubs-64.h instead
+          sed -i 's/#include <gnu\/stubs\.h>/\n/' /usr/include/features.h
+
+          # TODO: Alternatively, we could patch the unused variable out...
+          export CFLAGS="-O2 -Wall -fomit-frame-pointer -march=x86-64-v2 -Wp,-D_FORTIFY_SOURCE=3"
+          CFLAGS="$CFLAGS -Wno-error=unused-but-set-variable" make -C bpf-apache CC=clang LD=llc
+          CFLAGS="$CFLAGS -Wno-error=unused-but-set-variable" make -C bpf-gpl CC=clang LD=llc
+      # Equivalent to target: "libbpf"
+      - runs: |
+          _arch=$(uname -m); case $_arch in aarch64) _arch="arm64" ;; x86_64) _arch="amd64" ;; esac
+
+          mkdir -p bpf-gpl/include/libbpf/src/$_arch
+          export CFLAGS="-O2 -Wall -fomit-frame-pointer -march=x86-64-v2 -Wp,-D_FORTIFY_SOURCE=3"
+          CFLAGS="$CFLAGS" make -C bpf-gpl/include/libbpf/src BUILD_STATIC_ONLY=2 OBJDIR=$_arch
+  # This fetches and builds the statically compiled calico fork of BIRD
+  - working-directory: bird
+    pipeline:
+      - uses: fetch
+        with:
+          uri: https://github.com/projectcalico/bird/tarball/v0.3.3-202-g7a77fb73
```

Comment:
Is this updated, and do we need to account for updating it to pick up new releases?

Reply:
this still needs to be updated manually for now. in a future PR I'll make a `bird` package which we can leverage in `calico`. but as of today I don't know all the deltas between calico's fork.

since this changes out of band from calico proper (relevant [blame](https://github.com/projectcalico/calico/blame/e40707c5d30c353a78bed67a8c6596656b183121/metadata.mk#L26)), we hopefully have some time though